worker/docker_image_manager.py

```diff
@@ -0,0 +1,140 @@
+import threading
+import os
+import time
+import traceback
+import logging
+import json
+
+import file_util
+from formatting import size_str
+
+logger = logging.getLogger(__name__)
+
+
+class DockerImageManager(object):
+    """
+    Manages the Docker images available on the worker, and ensures that they
+    do not fill up the disk.
+
+    The last-used dates are only tracked for images on which
+    `DockerImageManager.touch_image()` has been called. All other images will
+    be left untouched by the cleanup thread.
+
+    DependencyManager should be instantiated before DockerImageManager, to
+    ensure that the work directory already exists.
+    """
+    STATE_FILENAME = 'images-state.json'
+
+    def __init__(self, docker, work_dir, min_free_bytes=1e9,
```

Comment:
Proposal: Maybe you can specify a `max_docker_images_cache_size`, and we will maintain at least `DISK_SIZE - (max_docker_images_cache_size + max_work_dir_size)` free on the disk. This has the potentially confusing side-effect of sometimes exceeding the `max_docker_images_cache_size` when the dependency cache is not full, but at least disk usage will not exceed `(max_docker_images_cache_size + max_work_dir_size)`. @percyliang thoughts?